log4j-api/src/main/java/org/apache/logging/log4j/CloseableThreadContext.java

```diff
@@ -74,6 +74,31 @@ private CloseableThreadContext() {
         return new CloseableThreadContext.Instance().put(key, value);
     }
 
+    /**
+     * Populates the Thread Context Stack with the supplied stack. The information will be popped off when
+     * the instance is closed.
+     *
+     * @param values The stack of values to be added
+     * @return a new instance that will back out the changes when closed.
+     * @since 2.7.1
+     */
+    public static CloseableThreadContext.Instance pushAll(final ThreadContext.ContextStack stack) {
```

Comment:
I was anticipating that the usage will be simply `ThreadContext.getImmutableStack()` but as you say `List` is probably easier (and more closely aligns with the `Map` used for `putAll(...)`).